arrow-glib: init at 8.0.0

amarshall opened the pull request:
###### Description of changes

This is necessary for some Arrow language libraries, notably the Ruby Arrow library ([red-arrow](https://rubygems.org/gems/red-arrow)).

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

amarshall committed:
arrow-cpp-glib: init at 8.0.0

veprbl commented on pkgs/top-level/all-packages.nix:
```diff
@@ -16957,6 +16957,8 @@ with pkgs;
 
   arrow-cpp = callPackage ../development/libraries/arrow-cpp {};
 
+  arrow-cpp-glib = callPackage ../development/libraries/arrow-cpp-glib {};
```
```suggestion
  arrow-glib = callPackage ../development/libraries/arrow-glib {};
```
https://arrow.apache.org/docs/c_glib/

veprbl replied:
Or `arrow-c-glib`

veprbl commented on pkgs/development/libraries/arrow-cpp-glib/default.nix:
```diff
@@ -0,0 +1,34 @@
+}:
+
+stdenv.mkDerivation rec {
+  pname = "${arrow-cpp.pname}-glib";
```
```suggestion
  pname = "arrow-glib";
```

amarshall commented:
Hi @veprbl! Thank you for the review. I have adjusted the package name as suggested.

amarshall committed:
arrow-cpp: Adjust homepage to specific variant

There are lots of Arrow implementations, so instead go to the one that
this implements.

veprbl commented:
The commit name should match the attribute name "arrow-glib: init at 8.0.0"

veprbl commented:
This can be used to enable tests
```diff
diff --git a/pkgs/development/libraries/arrow-cpp/default.nix b/pkgs/development/libraries/arrow-cpp/default.nix
index 64fc6a165c4..93ac8912438 100644
--- a/pkgs/development/libraries/arrow-cpp/default.nix
+++ b/pkgs/development/libraries/arrow-cpp/default.nix
@@ -2,6 +2,7 @@
 , lib
 , fetchurl
 , fetchFromGitHub
+, fetchpatch
 , fixDarwinDylibNames
 , abseil-cpp
 , autoconf
@@ -110,6 +111,14 @@ stdenv.mkDerivation rec {
   patches = [
     # patch to fix python-test
     ./darwin.patch
+
+    # fix some breakage caused by multiple-outputs cmake options
+    (fetchpatch {
+      url = "https://github.com/apache/arrow/commit/a242eb17362c0352fc3291213542c48abfe18669.diff";
+      hash = "sha256-y2YBHNaWtMy/LJXT83uJjekKvckABAibYbxJuRYjlrA=";
+      stripLen = 2;
+      extraPrefix = "";
+    })
   ];
 
   nativeBuildInputs = [
diff --git a/pkgs/development/libraries/arrow-glib/default.nix b/pkgs/development/libraries/arrow-glib/default.nix
index 9c396f89d11..84c15d43c26 100644
--- a/pkgs/development/libraries/arrow-glib/default.nix
+++ b/pkgs/development/libraries/arrow-glib/default.nix
@@ -7,6 +7,8 @@
 , meson
 , ninja
 , pkg-config
+, ruby
+, rubyPackages
 }:
 
 stdenv.mkDerivation rec {
@@ -26,6 +28,18 @@ stdenv.mkDerivation rec {
     gobject-introspection
   ];
 
+  mesonFlags = [
+    "--bindir=bin"
+  ];
+
+  doInstallCheck = true;
+  dontUseNinjaCheck = true;
+  installCheckPhase = "ninjaCheckPhase";
+  installCheckInputs = [
+    ruby
+    rubyPackages.gobject-introspection
+  ];
+
   meta = with lib; {
     inherit (arrow-cpp.meta) license platforms;
     description = "GLib bindings for Apache Arrow";

````

risicle reviewed (approved):
`nixpkgs-review` happy, macos 10.15 & nixos x86_64
